kernel/include/syscall_handler.h

```diff
@@ -17,6 +17,12 @@
 
 extern const _k_syscall_handler_t _k_syscall_table[K_SYSCALL_LIMIT];
 
+enum _obj_init_check {
```

Comment:
Since this is used inside the kernel object validation function, it might be a good idea to define these values to be -1, 0, and 1.  Integer comparison is already cheaper, but not only 0 is special, checking for the sign bit is cheaper. I'd still make `_OBJ_INIT_TRUE = 0`, as that's the one that's more likely to be true.

Reply:
done